test/new-e2e/pkg/utils/e2e/client/host_cache.go

```diff
@@ -0,0 +1,179 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package client
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/components"
+	oscomp "github.com/DataDog/test-infra-definitions/components/os"
+)
+
+type unimplementedHostCache struct{}
+
+func (c *unimplementedHostCache) Get(_ string, _ string) error {
+	return fmt.Errorf("not implemented")
+}
+
+func hostArtifactsClientFactory(sshExecutor *sshExecutor, osFlavor oscomp.Flavor, cloudProvider components.CloudProviderIdentifier, architecture oscomp.Architecture) HostArtifactClient {
+	archString := ""
+	switch architecture {
+	case oscomp.AMD64Arch:
+		archString = "x86_64"
+	case oscomp.ARM64Arch:
+		archString = "aarch64"
+	}
+	switch cloudProvider {
+	case components.CloudProviderAWS:
+		switch osFlavor {
+		case oscomp.Debian, oscomp.Ubuntu:
+			return &hostArtifactsClient{
+				cli: &unixAWSCLI{
+					sshExecutor: sshExecutor,
+					archString:  archString,
+					pkgManager: &aptPkgManager{
+						sshExecutor: sshExecutor,
+					},
+				},
+			}
+		case oscomp.AmazonLinux, oscomp.CentOS, oscomp.RedHat, oscomp.RockyLinux, oscomp.Fedora:
+			return &hostArtifactsClient{
+				cli: &unixAWSCLI{
+					sshExecutor: sshExecutor,
+					archString:  archString,
+					pkgManager: &yumPkgManager{
+						sshExecutor: sshExecutor,
+					},
+				},
+			}
+		case oscomp.Suse:
+			return &hostArtifactsClient{
+				cli: &unixAWSCLI{
+					sshExecutor: sshExecutor,
+					archString:  archString,
+					pkgManager: &zypperPkgManager{
+						sshExecutor: sshExecutor,
+					},
+				},
+			}
+		case oscomp.WindowsServer:
+			return &hostArtifactsClient{
+				cli: &windowsAWSCLI{
+					sshExecutor: sshExecutor,
+				},
+			}
+		default:
+			return &unimplementedHostCache{}
+		}
+	default:
+		return &unimplementedHostCache{}
+	}
+}
+
+type cli interface {
+	install() error
+	check() bool
+	download(path string, destPath string) error
+}
+
+type hostArtifactsClient struct {
+	cli cli
+}
+
+type pkgManager interface {
+	install(pkgName string) error
+}
+
+type windowsAWSCLI struct {
+	sshExecutor *sshExecutor
+}
+
+func (c *windowsAWSCLI) install() error {
+	_, err := c.sshExecutor.Execute("Start-Process msiexec.exe -Wait -ArgumentList \"/i https://awscli.amazonaws.com/AWSCLIV2.msi /qn /norestart /L*V ./awscli-install.log\" ")
+	return err
+}
+
+func (c *windowsAWSCLI) check() bool {
+	_, err := c.sshExecutor.Execute("& \"c:\\Program Files\\Amazon\\AWSCLIV2\\aws.exe\" --version")
+	return err == nil
+}
+
+func (c *windowsAWSCLI) download(path string, destPath string) error {
+	_, err := c.sshExecutor.Execute(fmt.Sprintf("& \"c:\\Program Files\\Amazon\\AWSCLIV2\\aws.exe\" s3 cp %s %s", path, destPath))
```

Comment:
I suppose it's safer like this:
```suggestion
	_, err := c.sshExecutor.Execute(fmt.Sprintf("& \"c:\\Program Files\\Amazon\\AWSCLIV2\\aws.exe\" s3 cp \"%s\" \"%s\"", path, destPath))
```